### PR TITLE
Cargo.toml: fix h2 vulnerability

### LINKED ISSUE #16405: cargo build fails due to vulnerability in h2
## Cargo build fails due to vulnerability in h2

```
error[vulnerability]: Resource exhaustion vulnerability in h2 may lead to Denial of Service (DoS)
    ┌─ /Users/alexf/repos/tikv/Cargo.lock:163:1
    │
163 │ h2 0.3.22 registry+https://github.com/rust-lang/crates.io-index
    │ --------------------------------------------------------------- security vulnerability detected
    │
    = ID: RUSTSEC-2024-0003
    = Advisory: https://rustsec.org/advisories/RUSTSEC-2024-0003
    = An attacker with an HTTP/2 connection to an affected endpoint can send a steady stream of invalid frames to force the
      generation of reset frames on the victim endpoint.
      By closing their recv window, the attacker could then force these resets to be queued in an unbounded fashion,
      resulting in Out Of Memory (OOM) and high CPU usage.
      
      This fix is corrected in [hyperium/h2#737](https://github.com/hyperium/h2/pull/737), which limits the total number of
      internal error resets emitted by default before the connection is closed.
    = Solution: Upgrade to ^0.3.24 OR >=0.4.2 (try `cargo update -p h2`)
    = h2 v0.3.22
```

### PR DESCRIPTION
**Pull request by afeinberg**
### What is changed and how it works?

Issue Number: Close #16405 

What's Changed:

```commit-message
Fix h2 vulnerability caught by `make clippy`.

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by bufferflies — APPROVED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -6957,7 +6957,7 @@ version = "1.6.3"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "97fee6b57c6a41524a810daee9286c02d7752c4253064d0b05472833a438f675"
 dependencies = [
- "cfg-if 1.0.0",
```
Comment (v01dstar) [inline Cargo.lock:6960]: Is this intended?

**Review by v01dstar — APPROVED** [review]
LGTM

File: Cargo.lock (hunk 1 above)
Comment (afeinberg) [reply Cargo.lock:6960]: i think it is a transitive dependency as part of updating h2.

**Pull request merged**